### PR TITLE
Restore elementwise min/max on SIMD, with explicit naming.

### PR DESCRIPTION
**Pull request by stephentyrone**
Having these as free functions causes expression too complex issues due to overload vis-a-vis min<Collection>. There are (at least) three plausible solutions:

1. Fix the typechecker. This is infeasable in the short term; or more precisely, we do not know how much work is involved.

2. Give these operations different names. Candidates discussed with core team include "pointwiseMin", "elementwiseMin", "lanewiseMin"; these all suffer from the flaw that when someone writes "min" in a SIMD context, they are essentially always looking for either the horizontal minimum (reduction) on a single vector or--more often--the lanewise minimum of two vectors (this operation). It would be odd to give the operation that people actually want the unnecessarily verbose name.

3. Make these operations static; this is, in effect, a different name, but it's one which frequently allows eliding the qualifier:

  let x = v + .min(v, w)

After considerable back-and-forth discussion with core, we settled on `pointwiseMin` and `pointwiseMax`. These aren't perfect, but they are explicit, which is worth a lot.

This rounds out SE-0251.

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci please test

**Comment by stephentyrone** [comment]
@swift-ci please test source compatibility

**Comment by stephentyrone** [comment]
@swift-ci please test source compatibility

File: stdlib/public/core/SIMDVector.swift [hunk 1]
```diff
@@ -1381,26 +1411,25 @@ where V: SIMD, V.Scalar: Comparable {
...
-    result[i] = V.Scalar.minimum(lhs[i], rhs[i])
+    result[i] = T.Scalar.minimum(a[i], b[i])
   }
   return result
 }
 
 /// The lanewise maximum of two vectors.
 ///
-/// Each element of the result is the maximum of the corresponding elements
+/// Each element of the result is the minimum of the corresponding elements
```
Comment (Dante-Broggi) [inline stdlib/public/core/SIMDVector.swift]: Copy paste error?
```suggestion
/// Each element of the result is the maximum of the corresponding elements
```

**Comment by stephentyrone** [comment]
@swift-ci please test

**Comment by stephentyrone** [comment]
Linux failure unrelated (base breakage).

**Comment by stephentyrone** [comment]
@swift-ci please smoke test linux

**Pull request merged**